### DISCUSSION TITLE
byte size of the response

### DISCUSSION THREAD

**Discussion by HassanIyan** [fastify/fastify#4627] [Q&A] [answered]
is there any way i can get and record the size of the response returned from a route?

**Comment by metcoder95** [comment]
Not at the moment. Though can be an interesting addition 🤔 

As a workaround, you can measure the byte length of your request on the `onSend` hook and attach it to the `reply` object, to later use it within the `onResponse` hook.

**Comment by Eomm** [comment, marked as answer]
Do you mean like this?

```js
const fastify = require('fastify')({ logger: !true });

fastify.get(
  '/',
  {
    onResponse: function hook(request, reply, done) {
      console.log(reply.raw.getHeader('Content-Length'));
      done();
    },
  },
  async (request, reply) => {
    return 'a'.repeat(42);
  }
);

fastify.inject('/').then((response) => {});
```

**Reply by metcoder95** [reply]
Ah, yes, pretty much. I was having in mind calculating the byte length with `Buffer` but totally missed we already do that for the `Content-Length` header. Thanks for the heads up 🙇
